.github/workflows/check-dist.pkl

```diff
@@ -0,0 +1,74 @@
+amends "package://pkg.pkl-lang.org/github.com/stefma/pkl-gha/com.github.action@0.0.3#/GitHubAction.pkl"
+
+// In TypeScript actions, `dist/` is a special directory. When you reference
+// an action with the `uses:` property, `dist/index.js` is the code that will be
+// run. For this project, the `dist/index.js` file is transpiled from other
+// source files. This workflow ensures the `dist/` directory contains the
+// expected transpiled code.
+//
+// If this workflow is run from a feature branch, it will act as an additional CI
+// check and fail if the checked-in `dist/` directory does not match what is
+// expected from the build.
+name = "Check Transpiled JavaScript"
+
+on {
+  pull_request {}
+  push {
+    branches {
+      "main"
+    }
+  }
+}
+
+permissions {
+  contents = "read"
+}
+
+jobs {
+  ["check-dist"] {
+    name = "Check Dist"
+    `runs-on` = "ubuntu-latest"
+    steps {
+      new {
+        name = "Checkout"
+        uses = "actions/checkout@v4"
+      }
+      new {
+        name = "Setup Node.js"
+        uses = "actions/setup-node@v4"
+        with {
+          ["node-version-file"] = ".nvmrc"
+          ["cache"] = "npm"
+        }
+      }
+      new {
+        name = "Install Dependencies"
+        run = "npm ci"
+      }
```

Comment:
A fun future improvement would be showing that this can be refactored into a utility module so that we have the same configuration for checkout and setting up node across all workflows.

Reply:
Done here 👉  https://github.com/pkl-community/setup-pkl/pull/47/commits/54eb10e28c653def5c95b8fd4dba0e869501b472